src/ir/term/ty.rs

```diff
@@ -128,6 +128,49 @@ pub fn check_raw(t: &Term) -> Result<Sort, TypeError> {
             }
         }
         Op::Update(_i) => Ok(check_raw(&t.cs[0])?),
+        Op::Map(op) => {
+            let arg_cnt = t.cs.len();
+            let mut dterm_cs = Vec::new();
+
+            let mut key_sort = Sort::Bool;
+            let mut size = 0;
+            let mut error = None;
+
+            match arrmap_or(&check_raw(&t.cs[0])?, "map") {
+                Ok((k, _, s)) => {
+                    key_sort = k.clone();
+                    size = s.clone();
+                }
+                Err(e) => {
+                    error = Some(e);
+                }
+            }
+
+            for i in 0..arg_cnt {
+                match array_or(&check_raw(&t.cs[i])?, "map inputs") {
```

Comment:
should we use `arrmap_or` here as well to check that all input array sizes are the same? Then we can also combine both for loops together.

Reply:
I think it would be better if I added the check for the input array sizes to rec_check_raw() instead, since
check_raw() is mostly used to retrieve the type of a term, while rec_check_raw() is used to verify that the terms are well typed.
Do think this would be better?

Also, should I check that the key types are consistent throughout the input arrays as well?